CRM/MembersOnlyEvent/Service/MembersOnlyEventAccess.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+use CRM_MembersOnlyEvent_BAO_EventMembershipType as EventMembershipType;
+use CRM_MembersOnlyEvent_Configurations as Configurations;
+use CRM_MembersOnlyEvent_BAO_MembersOnlyEvent as MembersOnlyEvent;
+
+class CRM_MembersOnlyEvent_Service_MembersOnlyEventAccess {
+
+  /**
+   * @var bool
+   */
+  private static $isBooted = FALSE;
+
+  /**
+   * @var array
+   */
+  private static $contactActiveAllowedMemberships = [];
+
+  /**
+   * @var \CRM_MembersOnlyEvent_DAO_MembersOnlyEvent|null
+   */
+  private static $membersOnlyEvent = NULL;
+
+  /**
+   * @var int|null
+   */
+  private $eventID = NULL;
+
+  /**
+   * @param bool $fresh
+   */
+  public function __construct($fresh = FALSE) {
```

Comment:
on a 2nd thought, using statics does not look neat, maybe let's not use them? 🦊

Reply:
I've dropped last commit